halo2/src/circuit_builder.rs

```diff
@@ -36,7 +36,7 @@ pub(crate) fn analyzed_to_circuit<T: FieldElement>(
 
     let query = |column, rotation| Expr::Var(PlonkVar::Query(ColumnQuery { column, rotation }));
 
-    let mut cd = CircuitData::from(fixed, witness, &analyzed.constants);
+    let mut cd = CircuitData::from(fixed.to_owned(), witness, &analyzed.constants);
```

Comment:
I tried, but `CircuitData` needs a `Vec` as it has to append to it.

If a `Vec` is needed inside a function, I have a preference on taking it directly as the argument.

Reply:
That's my point with the selectors: `CircuitData` needs to introduce 2 selectors `__enable_cur` and `__enable_next`. Instead of adding them to the fixed columns, we can just stored them in another struct field in `CircuitData`, then `fixed` doesn't need to be mutable. Then when hitting halo2 we can add these selectors: `fixed: .fixed.iter().chain(once(cd.sel0)).chain(cd.sel1)`. It's sad to clone the whole fixed data for 2 selectors.